pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1089,12 +1091,25 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
                 }
                 Optional<Map<String, String>> subscriptionProperties = SubscriptionOption.getPropertiesMap(
                         subscribe.getSubscriptionPropertiesList());
+
+                boolean createTopicIfDoesNotExist = forceTopicCreation || (isAuthorizedToCreateTopic
+                        && service.isAllowAutoTopicCreation(topicName.toString()));
```

Comment:
I think we want the following in order to retain the existing logic:

```suggestion
                boolean createTopicIfDoesNotExist = isAuthorizedToCreateTopic || (forceTopicCreation
                        && service.isAllowAutoTopicCreation(topicName.toString()));
```